### ISSUE TITLE
Application does not have keychain access groups enabled in the entitlements.plist

### ISSUE THREAD

**Issue by keerthinathganjam** [Azure-Samples/active-directory-xamarin-native-v2#157]
I am tying to run the sample for IOS, I am getting the error

![MicrosoftTeams-image (7)](https://user-images.githubusercontent.com/12542938/101870314-5f564a80-3ba7-11eb-96b4-cf1fac25fcea.png)

I have set the Entitlements.plist manually in the the IOS Project as said in the document https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-net-xamarin-ios-considerations#enable-keychain-access 

![image](https://user-images.githubusercontent.com/12542938/101871378-785ffb00-3ba9-11eb-8370-5f79578db1c2.png)

I have a apple developer account also, which i have connected through VS, am i missing someting here? 
Is there any configuration i need to make in apple app store like creating app or creating some resource?
Should it be deplyoyed through MDM like Intunes for making it work?
We have both developer and enterprice accounts with Apple. I have tried pointing to both Developer and Enterprice accounts for Signing, I have tried manual and automatic Provisioning. What ever is the case i am hitting at the same error. :(

Basically we are looking to intergrate MSAL in to our android and IOS Application, Where our customers want enable Conditional Access Which require Device and Platform Identification to be Enabled In Place.

Problem we are facing for IOS app is when user click on signinwithbroker, it is redirected to broker app i.e., Microsoft Authenticator. Once i select the userid to sign in it returns back to Sample AAD app where i am getting this Error.

Note: I was able to get it work for Android App and Get Information in AZure AD SignIn Log such as Device Info and User Agent and other things

**Comment by jennyf19** [comment]
@keerthinathganjam looks to be an issue w/the keychain and how things are set up w/Apple. I don't think we will be of much help here, the best we can do is the error message coming back, which is saying something is wrong w/the keychain access.

Here's a link to [Apple's developer](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps?language=objc) documentation about the keychain.

If you have the [entitlements set correctly](https://github.com/Azure-Samples/active-directory-xamarin-native-v2/blob/master/2-With-broker/UserDetailsClient/UserDetailsClient.iOS/Entitlements.plist#L5) and `.WithIosKeychainSecurityGroup("com.microsoft.adalcache")', as shown [here](https://github.com/Azure-Samples/active-directory-xamarin-native-v2/blob/master/2-With-broker/UserDetailsClient/UserDetailsClient/MainPage.xaml.cs#L39), there is not much more we can do on our end, as the sample is working. You'd have to check the Apple developer settings, make sure the right user is signed in to Visual Studio & the Mac and try things out. :(

**Comment by keerthinathganjam** [comment]
@jennyf19 is there anything we need to do specifically from apple portal side for this sample to be working? Like creating app or some other resources from the Apple Developer Portal?

**Comment by jennyf19** [comment]
@keerthinathganjam the sample is set up to work out of the box, but does require the apple developer account, and MSAL will pick up the TeamId by default, which is what you are seeing in the error message.

**Comment by keerthinathganjam** [comment]
@jennyf19 : If this has to work out of the box without any interverntion from apple developer portal configuration, i dont know how apple would be able to assist us :(. Is this sample for IOS is working in your environments?

**Comment by keerthinathganjam** [comment]
@jennyf19 : I am running this sample with Visual Studio 2019 on windows 10 and connecting a apple iPad [Deploying with help of HotReload]. is that fine?
Or is this has to be run only on Visual Studio For Mac??

**Comment by jennyf19** [comment]
@keerthinathganjam you can run it on a Mac or you connect to a Mac or Mac mini, which has XCode and the provisioning profiles and emulators. This is probably why the keychain error is popping up because you don't have provisioning certs downloaded onto the Mac, which XCode will pick up. @trwalke can maybe provide more guidance.

**Comment by trwalke** [comment]
@keerthinathganjam for more info about provisioning, you have 2 options.
1) Manual provisioning: this requires an apple developer account with a device that has been registered to run with the associated developer cert
[Documentation](https://docs.microsoft.com/en-us/xamarin/ios/get-started/installation/device-provisioning/manual-provisioning?tabs=windows)

2) Automatic provisioning: Should be much easier to run on an emulator with an apple developer account but a device can be used too. the provisioning configuration will be automatic
[Documentation](https://docs.microsoft.com/en-us/xamarin/ios/get-started/installation/device-provisioning/automatic-provisioning?tabs=windows)

**Comment by keerthinathganjam** [comment]
@jennyf19 : This sample for IOS works only Visual Studio On mac On a Mac Machine where XCode is installed. I have been trying this on VIsualstudio 2019 on windows. Now after using Visual studio for MAC on a Mac machine where xcode is installed its working now. Please update this point in the readme file / Documentation.

**Comment by jmprieur** [comment]
Thanks for the feedback, @keerthinathganjam 

I believe that this was explained in the section [Setup Xamarin.iOS for Visual Studio](https://docs.microsoft.com/en-us/xamarin/ios/get-started/installation/windows/introduction-to-xamarin-ios-for-visual-studio) (if you want to run the iOS app), but we've added a sentence to make it even clearer.
 
Closing this issue as the question is answerered

**Comment by keerthinathganjam** [comment]
@jennyf19 : This Works on on mac machine with visual studio on mac installed in that. It dosent work on windows machine with visual studio 2019 even with having an active connection to mac host.

**Comment by jmprieur** [comment]
@keerthinathganjam thanks for the update. That's a question for Visual Studio. You might want to try https://visualstudio.microsoft.com/vs/support/
